data/movements/movements.xml

```diff
@@ -35,95 +35,185 @@
 
 	<!-- (Level & quest) doors -->
 	<movevent event="StepOut" itemid="1224" script="closingdoor.lua" />
+	<movevent event="StepIn" itemid="1224" script="opendoor.lua" />
```

Comment:
Yes. Split level doors and quest doors into two different script files, then you don't need the isInArray check.